pkg/cache/lrucache.go

```diff
@@ -59,6 +60,19 @@ func NewGitRepoLRUCache(dir string, minFreeGbs uint64) (*GitRepoLRUCache, error)
 		return nil, fmt.Errorf("error checking provided cache directory: %s", err.Error())
 	}
 
+	stats := &syscall.Statfs_t{}
+
+	err = syscall.Statfs(dir, stats)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching stats for cache directory: %s", err.Error())	
+	} 
+
+	freeSpace := stats.Bavail * stats.Bfree
```

Comment:
Can we explicitly cast the `stats.Bfree` as a `uint64`?

```go
freeSpace := stats.Bavail * uint64(stats.Bfree)
```

The type from this value is derived from the `go/src/syscall/ztypes_{system_type}_{system_arch_bit_length}.go` package which will differ if compiled on 32 bit (or smaller) architectures.

On my macbook, it gets this `uint64` from `/usr/local/go/src/syscall/ztypes_darwin_arm64.go`.

This is all fine for now since I'm not worried about deploying this to anything besides 64 bit systems and we have a tracking issue for getting feedback on deploying to other arches / systems / bit-lengths: #23 

See the `tryEvict` function for an example:
 https://github.com/open-sauced/pizza/blob/51ed9ca9798ad3f9c39fc02a368fc1f525949737/pkg/cache/lrucache.go#L196-L199